docs/modules/migrate/pages/upgrading-from-jet.adoc

```diff
@@ -0,0 +1,197 @@
+[cols="1a,1m"]
+|===
+|JetInstance Usage|New class
+
+| Submitting streaming/batch jobs to the cluster and managing them.
+|JetService
+
+|Accessing Hazelcast data structures.
+a|`HazelcastInstance`
+
+The only exception is Jet observables. An observable is a Jet data structure and we ported it to the `JetService` class. 
+
+|Performing cluster operations such as shutting down the cluster.
+|HazelcastInstance
+|===
+
+To access Jet related services, you should now use the link:https://docs.hazelcast.org/docs/{page-component-version}/javadoc/com/hazelcast/core/HazelcastInstance.html#getJet--[`HazelcastInstance.getJet()` method] to get an instance of the `JetService` object.
+
+[tabs] 
+==== 
+Jet 4.x:: 
++ 
+-- 
+```java
+HazelcastInstance hz = Hazelcast.newHazelcastInstance();
+JetInstance jet = hz.getJetInstance();
+```
+--
+Hazelcast:: 
++ 
+-- 
+```java
+HazelcastInstance hz = Hazelcast.newHazelcastInstance();
+JetService jet = hz.getJet();
+```
+--
+====
+
+== Enabling the Jet Engine
+
+In embedded mode (Hazelcast as a classic Java process), Jet is disabled by default. As a result, you cannot use Jet APIs or SQL unless you enable the Jet engine.
+
+See xref:pipelines:job-security.adoc[].
+
+== Code Samples
+
+Jet code samples are now in the Hazelcast code samples repository. See link:https://github.com/hazelcast/hazelcast-code-samples/tree/master/jet[GitHub].
+
+== Version Compatibility
+
+Hazelcast {page-component-version} uses https://semver.org/[semantic versioning]:
+
+* MAJOR version when you make incompatible API changes
+* MINOR version when you add functionality in a backwards-compatible manner
+* PATCH version when you make backwards-compatible issue fixes.
+
+Hazelcast {page-component-version} is compatible only with the Job and Pipeline APIs of Jet 4.x. As a result, jobs that were running on Jet 4.x clusters can still be executed on Hazelcast {page-component-version} clusters.
+
+Hazelcast {page-component-version} *is not* compatible with any other aspects of Jet 4.x. You cannot upgrade using rolling upgrades and you cannot connect to the cluster using version 4.x of clients).
+
+Hazelcast {page-component-version} is not compatible with versions of Management Center before {page-component-version}.
+
+== Distributions
+
+Jet 4.x came in `slim` and `fat` distributions. Hazelcast {page-component-version} is now available in the following distributions:
+
+- Slim
+** Hazelcast runtime
+** Default configuration files (not example ones)
+** Scripts
+
+- Full
+
+** Hazelcast and its modules contained in the former `hazelcast-all`
+** AWS discovery plugin
+** Kubernetes discovery plugin
+** GCP discovery plugin
+** Azure discovery plugin
+** Hazelcast Hibernate
+** Hazelcast Spring
+** Default configuration files, including examples
+** Scripts
+** SQL
+** Jet extensions
+** Management center
+
+For a complete list of contents in each distribution, see xref:deploy:installing-upgrading.adoc#full-and-slim-packages[Full and Slim Packages].
```

Comment:
Please provide an action point. Does the distribution change require my attention?